src/plugins/intel_gna/gna_graph_compiler.cpp

```diff
@@ -194,6 +193,31 @@ void GNAGraphCompiler::fillSplitConnections(InferenceEngine::CNNLayerPtr layer)
     split_connection.emplace(id, layerInfoItem);
 }
 
+void GNAPluginNS::GNAGraphCompiler::SetValidatorTarget(std::string target) {
+    if (InferenceEngine::GNAConfigParams::GNA_TARGET_3_0 == target) {
+        cnn2dValidator.reset(new GNALimitations::Cnn2D::Validator_30());
+    }
+}
+
+void GNAPluginNS::GNAGraphCompiler::ValidateCnn2D(std::string name, const uint32_t inHeight, const uint32_t inWidth, const uint32_t inChannels,
+    const uint32_t kH, const uint32_t kW, const uint32_t kN, const uint32_t strideH, const uint32_t strideW, OvGnaType inPrecision,
+    const uint32_t dilH, const uint32_t dilW) const {
+    if (cnn2dValidator) {
+        cnn2dValidator->ValidateCnn2D(name, inHeight, inWidth, inChannels, kH, kW, kN, strideH, strideW, dilH, dilW, inPrecision);
+    } else {
+        THROW_GNA_EXCEPTION << "No CNN2D validator found for current target in layer " << name;
+    }
+}
+
+void GNAPluginNS::GNAGraphCompiler::ValidatePooling2D(std::string name, const uint32_t windowH, const uint32_t windowW,
+    const uint32_t strideH, const uint32_t strideW) const {
+    if (cnn2dValidator) {
+        cnn2dValidator->ValidatePooling2D(name, windowH, windowW, strideH, strideW);
+    } else {
+        THROW_GNA_EXCEPTION << "No CNN2D validator found for current target in layer " << name;
```

Comment:
It would be good to have a different message here than for the previous case.

Reply:
OK